cmake/common/check_configuration.cmake

```diff
@@ -130,7 +130,7 @@ function(set_common_compile_options target)
             -Wsign-conversion
             $<$<CXX_COMPILER_ID:GNU>:-Wlogical-op>
             $<$<AND:$<CXX_COMPILER_ID:GNU>,$<COMPILE_LANGUAGE:CXX>>:-Wuseless-cast>
-            -Wdouble-promotion
+            $<$<OR:$<AND:$<CXX_COMPILER_ID:GNU>,$<NOT:$<VERSION_LESS:$<CXX_COMPILER_VERSION>,4.6.0>>>,$<AND:$<C_COMPILER_ID:GNU>,$<NOT:$<VERSION_LESS:$<C_COMPILER_VERSION>,4.6.0>>>>:-Wdouble-promotion>
```

Comment:
Can you move this line after line 138 in order to keep the last conditional flag added to the last one?